projects/wvr-elements/src/lib/core/wvr-animation.service.ts

```diff
@@ -0,0 +1,107 @@
+import { animation, AnimationBuilder, AnimationMetadata, AnimationPlayer, AnimationReferenceMetadata, useAnimation } from '@angular/animations';
+import { Injectable } from '@angular/core';
+import { wvrAnimationDefaults, wvrAnimationInitialization, wvrAnimations } from '../shared/animation/wvr-animations';
+import { WvrBaseComponent } from '../shared/wvr-base.component';
+
+@Injectable({
+  providedIn: 'root',
+  deps: [AnimationBuilder]
+})
+export class WvrAnimationService {
+
+  private recieversRegistry = new Map<string, Array<WvrBaseComponent>>();
+
+  private animationStates = new Map<number, Map<string, boolean>>();
+
+  constructor(private readonly builder: AnimationBuilder) { }
+
+  registerAnimationReciever(recieverName: string, component: WvrBaseComponent): void {
+    let recievers = this.recieversRegistry.get(recieverName);
+    if (!recievers) {
+      recievers = new Array<WvrBaseComponent>();
+      this.recieversRegistry.set(recieverName, recievers);
+    }
+    recievers.push(component);
+  }
+
+  registerAnimationStates(): number {
+    const id = Math.random();
+    this.animationStates.set(id, new Map<string, boolean>());
+
+    return id;
+
+  }
+
+  triggerAnimationReciever(recieverName: string): void {
+    const recievers = this.recieversRegistry.get(recieverName);
+    if (recievers) {
+      recievers.forEach(r => r.triggerAnimations('animationTrigger'));
+    }
+  }
+
+  initializeAnimationElement(stateId, animationConfig: {}, animationRootElem): void {
+    Object.keys(animationConfig)
+      .forEach(animName => {
+        const animConf = animationConfig[animName];
+        const initializationMethod = wvrAnimationInitialization[animName];
+        if (initializationMethod && animationRootElem) {
+          const from = animConf.from ?
+                      animConf.from :
+                      wvrAnimationDefaults[animName].from;
+          initializationMethod(this.animationStates.get(stateId), from, animationRootElem.nativeElement);
+        }
+      });
+  }
+
+  compileAnimation(stateId, animationName, value): AnimationMetadata | Array<AnimationMetadata> {
```

Comment:
I recommend reordering these methods for better readability. `compileAnimation` Is called from `selectAnimation`, therefor should be after it. Similarly, `selectAnimation` Is called from `playAnimation`, therefor should be after it.